py/jpy-integration/build.gradle

```diff
@@ -82,54 +83,129 @@ idea {
   }
 }
 
-// only enable this if py building is enabled
-if (PyEnv.pythonEnabled(project)) {
+Task testJavaToPython = tasks.create 'testJavaToPython', {
+  Task t ->
+    t.group = 'python'
+    t.description = 'Run the java test suite that depends on deephaven-jpy (org.jpy.PyLib, etc)'
+    check.dependsOn t
+}
 
-  PyEnv env = PyEnv.getEnv(project)
+tasks.create 'debugJavaToPython', {
+  Task t ->
+    t.group = 'python'
+    t.description = 'Run the java exec io.deephaven.jpy.integration.PyDebug which is a sanity test for all JavaExec tasks'
+}
 
-  List<Task> pythonTests = []
-  Task testJavaToPython = tasks.create 'testJavaToPython', {
-    Task t ->
-      t.group = 'python'
-      t.description = 'Run the java test suite that depends on deephaven-jpy (org.jpy.PyLib, etc)'
-      check.dependsOn t
-      pythonTests += t
-  }
+tasks.create 'pathJavaToPython', {
+  Task t ->
+    t.group = 'python'
+    t.description = 'Run the java exec io.deephaven.jpy.integration.PySysPath which prints java properties and python paths'
+}
 
-  tasks.create 'debugJavaToPython', {
-    Task t ->
-      t.group = 'python'
-      t.description = 'Run the java exec io.deephaven.jpy.integration.PyDebug which is a sanity test for all JavaExec tasks'
-  }
+Task pythonToJava = tasks.create 'unittestPythonToJava', {
+  Task t ->
+    t.group = 'python'
+    t.description = 'Run "python -m xmlrunner discover" in src/python-to-java/python'
+    check.dependsOn t
+}
 
-  tasks.create 'pathJavaToPython', {
-    Task t ->
-      t.group = 'python'
-      t.description = 'Run the java exec io.deephaven.jpy.integration.PySysPath which prints java properties and python paths'
-  }
+if (!PyEnv.pythonEnabled(project)) {
+//  Task testTask = venvTest.javaTest(project, "java-to-python-test-${pv.name}", installWheels, sourceSets.javaToPython)
+//  testJavaToPython.dependsOn testTask
+
+  Closure<TaskProvider<Task>> javaMainInDocker = { String taskName, String javaMain, SourceSet sourceSet, List<String> jvmArgs = [] ->
+    return Docker.registerDockerTask(project, taskName) {
+      copyIn {
+        from(sourceSet.runtimeClasspath) {
+          into 'classpath'
+        }
+      }
+      parentContainers = [project(':Integrations').tasks.findByName('buildDeephavenPython')] // deephaven/runtime-base
+      imageName = 'deephaven/jpy-integration-java-to-python-tests'
+      dockerfile {
+        from 'deephaven/runtime-base'
 
-  tasks.create 'debugPythonToJava', {
-    Task t ->
-      t.group = 'python'
-      t.description = 'Run the python exec python/java_debug.py which is a sanity test for all python Exec tasks in this project'
+        copyFile 'classpath', '/classpath'
```

Comment:
Probably should have caught in earlier review, but is `copyFile` not the best name since we are coping the full directory?

Reply:
It isn't our name, it is part of the docker gradle plugin